lib/api.js

```diff
@@ -135,34 +138,57 @@ module.exports.multipartUpload = async function (options, uploadId, fileName, fi
   let partSize = Math.ceil(fileSize / MAX_PARTS);
   partSize = Math.max(partSize, FIVE_MB);
   const totalParts = Math.ceil(fileSize / partSize);
+  debug(`Uploading ${totalParts} parts of size ${partSize}`);
 
   for (let part = 1; part <= totalParts; ++part) {
     q.push(async () => {
       const start = (part - 1) * (partSize);
       const end = part === totalParts ? fileSize - 1 : start + partSize - 1;
       const size = end - start + 1;
-      // eslint-disable-next-line security/detect-non-literal-fs-filename
-      const stream = fs.createReadStream(fileName, {
-        start: start,
-        end: end
-      });
-
-      const response = await request(options).get(`/v1/uploads/${uploadId}/parts/${part}`);
-
-      await axios({
-        method: 'put',
-        url: response.data.uploadUrl,
-        data: stream,
-        headers: {
-          'Content-Length': size
+      let attempts = 0;
+
+      do {
+        // eslint-disable-next-line security/detect-non-literal-fs-filename
+        const stream = fs.createReadStream(fileName, {
+          start: start,
+          end: end
+        });
+
+        const response = await request(options).get(`/v1/uploads/${uploadId}/parts/${part}`);
+
+        try {
+          await axios({
+            method: 'put',
+            url: response.data.uploadUrl,
+            data: stream,
+            headers: {
+              'Content-Length': size
+            }
+          });
+          bar.increment(size);
+          debug(`Completed part ${part}, size: ${size}`);
+          return;
+        } catch (err) {
+          // S3 will respond with a 400 for a socket timeout which could occur from a slow stream read.
+          // Retry these errors up to 10 times to get the part uploaded.
+          if (err.response.status === 400) {
```

Comment:
Had to do manual retry logic here instead of relying on axiosRetry since a stream cannot be reset.